core/storage/blog_post/gae_models.py

```diff
@@ -0,0 +1,457 @@
+    """
+
+    # The user_id of the blog editors of this blog post.
+    editor_ids = datastore_services.StringProperty(indexed=True, repeated=True)
+
+    # Whether this blog post is published or not.
+    # False if blog post is a draft, True if published.
+    blog_post_is_published = datastore_services.BooleanProperty(
+        indexed=True, required=True, default=False)
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model contains data to pseudonymize or delete corresponding
+        to a user: editor_ids field.
+        """
+        return base_models.DELETION_POLICY.LOCALLY_PSEUDONYMIZE
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether BlogPostRightsModel references user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.query(
+            # NOTE: Even though `editor_ids` is repeated, we can compare it to a
+            # single value and it will return models where any of the editor IDs
+            # are equal to user_id.
+            cls.editor_ids == user_id
+        ).get(keys_only=True) is not None
+
+    @classmethod
+    def get_by_user(cls, user_id):
+        """Retrieves the rights object for all blog posts assigned to given user
+
+        Args:
+            user_id: str. ID of user.
+
+        Returns:
+            list(BlogPostRightsModel). The list of BlogPostRightsModel objects
+            in which the given user is a editor.
+        """
+
+        return cls.query(cls.editor_ids == user_id).fetch()
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model is exported as one instance shared across users since multiple
+        users can edit the blog post.
+        """
+        return (
+            base_models.MODEL_ASSOCIATION_TO_USER
+            .ONE_INSTANCE_SHARED_ACROSS_USERS
+        )
+
+    @classmethod
+    def get_export_policy(cls):
+        """Model contains data to export corresponding to a user."""
+        return dict(super(BlogPostRightsModel, cls).get_export_policy(), **{
+            'editor_ids': base_models.EXPORT_POLICY.EXPORTED,
+            'blog_post_is_published': base_models.EXPORT_POLICY.NOT_APPLICABLE
+        })
+
+    @classmethod
+    def export_data(cls, user_id):
+        """(Takeout) Export user-relevant properties of BlogPostsRightsModel.
+
+        Args:
+            user_id: str. The user_id denotes which user's data to extract.
+
+        Returns:
+            dict. The user-relevant properties of BlogPostsRightsModel.
+            in a python dict format. In this case, we are returning all the
+            ids of blog posts for which the user is editor of.
+        """
+        editable_blog_posts = cls.get_all().filter(cls.editor_ids == user_id)
+
+        editable_blog_post_ids = [blog.key.id() for blog in editable_blog_posts]
```

Comment:
blog.id suffices, right?

Reply:
I tried using blog.id but it gives errors as we receive a query string `Query(kind='BlogPostRightsModel', filters=AND(FilterNode('deleted', '=', False), FilterNode('editor_ids', '=', 'id_x')))` using `get_all().filter`.
We then use blog.key to receive the key of the model like `Key('BlogPostRightsModel', 'blog_post_old_id')` and then we use `.id()` to get its id.

Reply:
That's probably because you might have forgotten to add .fetch() after the filter(). Once you do that, `.id` should work.

Reply:
I don't want to fetch the complete model as I just want the id. So I did not add .fetch()

Reply:
Look into "projection queries". Can they help you here?

Reply:
Even after using projection query we will have to use.key as 'id' for models is always stored as a part of key and not as separate property.
The model I get after performing fetch() is : `BlogPostRightsModel(key=Key('BlogPostRightsModel', 'blog_post_id'), blog_post_is_published=True, created_on=datetime.datetime(2021, 6, 13, 10, 27, 32, 606512), deleted=False, editor_ids=[u'user_2'], last_updated=datetime.datetime(2021, 6, 13, 10, 27, 32, 606518))`
So, we will have to use `.key`

Reply:
OK, here's the thing -- pretty much nothing else in the codebase uses key.id, generally speaking (I note a few exceptions that have crept in). So, it would be better to standardize.

I don't think what you're saying is correct, did you actually try out using `id`? See e.g. base_model/gae_models.py and take a look at export_data() for the BaseSnapshotMetadataModel. Note that `id` is defined as you have it in BaseModel.

Reply:
Ah sorry, I was calling `id()` in `blog.id`. So it was giving errors. Fixed.